doc/man3/BIO_ctrl.pod

```diff
@@ -92,7 +92,7 @@ for success and -1 for failure.
 
 BIO_flush() returns 1 for success and 0 or -1 for failure.
 
-BIO_eof() returns 1 if EOF has been reached 0 otherwise.
+BIO_eof() returns 1 if EOF has been reached, 0 if not, or < 0 on error.
```

Comment:
No! This is wrong, you cannot change public API like this.

Reply:
Same as above.

Reply:
As I said above this can happen only on _impossible_ errors such as BIO_CTRL_EOF or bio_ctrl call not implemented. I'd then recommend to change the BIO_eof() code to return 1 in these cases instead of changing all the callers.

Reply:
Yeah, let's do the correction at `BIO_eof()` level.

Reply:
Done